crates/primitives/src/transaction/mod.rs

```diff
@@ -531,6 +594,31 @@ impl Transaction {
                 input.0.encode(out);
                 access_list.encode(out);
             }
+            Transaction::Eip4844(TxEip4844 {
+                chain_id,
+                nonce,
+                gas_limit,
+                max_fee_per_gas,
+                max_priority_fee_per_gas,
+                to,
+                value,
+                access_list,
+                blob_versioned_hashes,
+                max_fee_per_blob_gas,
+                input,
+            }) => {
+                chain_id.encode(out);
+                nonce.encode(out);
+                gas_limit.encode(out);
+                max_fee_per_gas.encode(out);
+                max_priority_fee_per_gas.encode(out);
+                to.encode(out);
+                value.encode(out);
+                access_list.encode(out);
+                blob_versioned_hashes.encode(out);
+                max_fee_per_blob_gas.encode(out);
+                input.0.encode(out);
```

Comment:
the eip defines the order as:

[chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, to, value, data, access_list, max_fee_per_blob_gas, blob_versioned_hashes, y_parity, r, s]

https://eips.ethereum.org/EIPS/eip-4844

similar to eip1559 tx type, see above

Reply:
@mattsse Sure you are right, I've just fixed that.